marlowe-chain-sync/src/Language/Marlowe/Runtime/ChainSync/NodeClient.hs

```diff
@@ -244,7 +252,16 @@ mkClientStNext costModel maxCost changesVar getHeaderAtPoint pipelineDecision n
       let clientTip = At blockNo
       pure $ mkClientStIdle costModel maxCost changesVar getHeaderAtPoint pipelineDecision n clientTip tip
   , recvMsgRollBackward = \point tip -> do
-      clientTip <- fmap blockHeaderToBlockNo <$> runGetHeaderAtPoint getHeaderAtPoint point
+      tipFromChanges <- atomically $ lookupPointInChanges point <$> readTVar changesVar
+      clientTip <- case tipFromChanges of
+        Nothing -> do
+          mClientTip <- (fmap . fmap) blockHeaderToBlockNo <$> runGetHeaderAtPoint getHeaderAtPoint point
+          case mClientTip of
+            Nothing -> error $ "Failed to process rollback. Failed to load block number for rollback point " <> show point
+            Just clintTip -> pure clintTip
+        Just clintTip -> do
+          hPutStrLn stderr "Found rollback point in node client changes"
```

Comment:
. . . and if there is a rollback that isn't rolled back, we would receive this message?

Reply:
Not necessarily. We'd see this message if we see a rollback while the chain store is persisting a batch of changes, and the node client has a collection of blocks staged for the store to save when it is available.